shared/common-adapters/team-with-popup/index.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react'
+import {Box2} from '../box'
+import Text from '../text'
+import * as Styles from '../../styles'
+import {TextType} from '../text'
+import TeamInfo from '../../profile/user/teams/teaminfo'
+import DelayedMounting from '../delayed-mounting'
+
+const Kb = {
+  Box2,
+  Text,
+}
+export type Props = {
+  description: string
+  isMember: boolean
+  isOpen: boolean
+  inline?: boolean
+  loadTeam: () => void
+  memberCount: number
+  onJoinTeam: () => void
+  onViewTeam: () => void
+  prefix?: string
+  teamName: string
+  type: TextType
+  underline?: boolean
+}
+
+export const TeamWithPopup = (props: Props) => {
+  const {loadTeam, onJoinTeam, onViewTeam} = props
+  const {description, isMember, isOpen, memberCount} = props
+  const {prefix, teamName, type, inline} = props
+  const popupRef = React.useRef(null)
+  const [showPopup, setShowPopup] = React.useState(false)
+  // load team details once on mount
```

Comment:
Can you pass a prop as to whether we actually need to do this? I know in the chat sidebar bot case we do, but in a lot of the app we're working on knowing whether we're interested in a team and keeping it up to date if so. So in those cases (e.g. if this component were to appear on the teams tab) we likely wouldn't need this extra load.

Reply:
Sure, makes sense